deliver/lib/deliver/submit_for_review.rb

```diff
@@ -41,17 +42,17 @@ def select_build(options)
         end
       else
         UI.message("Selecting the latest build...")
-        build = wait_for_build(app)
+        build = wait_for_build(app, app_version)
       end
-      UI.message("Selecting build #{build.train_version} (#{build.build_version})...")
+      UI.message("Selecting build #{app_version} (#{build.build_version})...")
```

Comment:
That is incorrect, the `app_version` passed to `deliver` is `build_version` not `train_version`.